pkg/cmd/cli/secrets/basicauth.go

```diff
@@ -183,22 +197,8 @@ func (o CreateBasicAuthSecretOptions) Validate() error {
 		return errors.New("basic authentication secret name must be present")
 	}
 
-	if o.PromptForPassword {
-		if len(o.Password) != 0 {
-			return errors.New("must provide either --prompt or --password flag")
-		}
-		if cmdutil.IsTerminal(o.Reader) {
-			o.Password = cmdutil.PromptForPasswordString(o.Reader, o.Out, "Password: ")
-			if len(o.Password) == 0 {
-				return errors.New("password must be provided")
-			}
-		} else {
-			return errors.New("provided reader is not a terminal")
-		}
-	} else {
-		if len(o.Username) == 0 && len(o.Password) == 0 && len(o.CertificatePath) == 0 {
-			return errors.New("must provide basic authentication credentials")
-		}
+	if len(o.Username) == 0 && len(o.Password) == 0 {
```

Comment:
@csrwng based on the https://github.com/openshift/origin/pull/4433#discussion-diff-38798689 conversation, we need at least one of those credentials.